lnwallet/chancloser/chancloser_test.go

```diff
@@ -377,3 +464,131 @@ func TestParseUpfrontShutdownAddress(t *testing.T) {
+			},
+			FeeEstimator: &SimpleCoopFeeEstimator{},
+			DisableChannel: func(wire.OutPoint) error {
+				return nil
+			},
+		}, nil, idealFee, 0, nil, false,
+	)
+	bobCloser.initFeeBaseline()
+
+	// With our set up complete, we'll now initialize the shutdown
+	// procedure kicked off by Alice.
+	msg, err := aliceCloser.ShutdownChan()
+	require.NoError(t, err)
+	require.NotNil(t, msg)
+
+	// Bob will then process this message. As he's the responder, he should
+	// only send the shutdown message back to Alice.
+	bobMsgs, closeFinished, err := bobCloser.ProcessCloseMsg(msg)
+	require.NoError(t, err)
+	require.False(t, closeFinished)
+	require.Len(t, bobMsgs, 1)
+	require.IsType(t, &lnwire.Shutdown{}, bobMsgs[0])
+
+	// Alice should process the shutdown message, and create a closing
+	// signed of her own.
+	aliceMsgs, closeFinished, err := aliceCloser.ProcessCloseMsg(bobMsgs[0])
+	require.NoError(t, err)
+	require.False(t, closeFinished)
+	require.Len(t, aliceMsgs, 1)
+	require.IsType(t, &lnwire.ClosingSigned{}, aliceMsgs[0])
+
+	// Next, Bob will process the closing signed message, and send back a
+	// new one that should match exactly the offer Alice sent.
+	bobMsgs, closeFinished, err = bobCloser.ProcessCloseMsg(aliceMsgs[0])
+	require.NoError(t, err)
+	require.True(t, closeFinished)
+	require.Len(t, aliceMsgs, 1)
+	require.IsType(t, &lnwire.ClosingSigned{}, bobMsgs[0])
+
+	// At this point, Bob has accepted the offer, so he can broadcast the
+	// closing transaction, and considers the channel closed.
+	_, err = lnutils.RecvOrTimeout(broadcastSignal, time.Second*1)
+	require.NoError(t, err)
+
+	// Bob's fee proposal should exactly match Alice's initial fee.
+	aliceOffer := assertType[*lnwire.ClosingSigned](t, aliceMsgs[0])
+	bobOffer := assertType[*lnwire.ClosingSigned](t, bobMsgs[0])
+	require.Equal(t, aliceOffer.FeeSatoshis, bobOffer.FeeSatoshis)
+
+	// If we modify Bob's offer, and try to have Alice process it, then she
+	// should reject it.
+	ogOffer := bobOffer.FeeSatoshis
+	bobOffer.FeeSatoshis /= 2
+
+	aliceMsgs, _, err = aliceCloser.ProcessCloseMsg(bobOffer)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "was not accepted")
+
+	// We'll now restore the original offer before passing it on to Alice.
+	bobOffer.FeeSatoshis = ogOffer
+
+	// If we use the original offer, then Alice should accept this message,
+	// and finalize the shutdown process. We expect a message here as Alice
+	// will echo back the final message.
+	aliceMsgs, closeFinished, err = aliceCloser.ProcessCloseMsg(bobMsgs[0])
+	require.NoError(t, err)
+	require.True(t, closeFinished)
+	require.Len(t, aliceMsgs, 1)
+	require.IsType(t, &lnwire.ClosingSigned{}, aliceMsgs[0])
+
+	// Alice should now also broadcast her closing transaction.
+	_, err = lnutils.RecvOrTimeout(broadcastSignal, time.Second*1)
+	require.NoError(t, err)
+
+	// Finally, Bob will process Alice's echo message, and then also
+	// attempt to conclude and broadcast the transaction himself.
```

Comment:
comment is wrong since Bob is completely done now and is in state `closeFinished`